Golem.Tests/ProviderFailureHandlingTest.cs

```diff
@@ -0,0 +1,131 @@
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading.Channels;
+using System;
+using System.Management;
+
+using Golem.Tools;
+using Golem.Yagna.Types;
+
+using GolemLib;
+using GolemLib.Types;
+
+using Microsoft.Extensions.Logging;
+using System.Runtime.InteropServices;
+
+namespace Golem.Tests
+{
+    [Collection(nameof(SerialTestCollection))]
+    public class ErrorHandlingTests : JobsTestBase
+    {
+
+        public ErrorHandlingTests(ITestOutputHelper outputHelper, GolemFixture golemFixture)
+            : base(outputHelper, golemFixture)
+        { }
+
+        [OsFact(nameof(OSPlatform.Windows))]
+        public async Task StartTriggerErrorStop_VerifyStatusAsync()
+        {
+            // Having
+            var logger = _loggerFactory.CreateLogger(nameof(StartTriggerErrorStop_VerifyStatusAsync));
+
+            string golemPath = await PackageBuilder.BuildTestDirectory();
+            logger.LogInformation("Path: " + golemPath);
+
+            var golem = await TestUtils.Golem(golemPath, _loggerFactory);
+            var statusChannel = TestUtils.PropertyChangeChannel<Golem, GolemStatus?>((Golem)golem, nameof(Golem.Status), _loggerFactory);
+
+            var jobStatusChannel = PropertyChangeChannel<IJob, JobStatus>(null, "");
+
+            Channel<Job?> jobChannel = PropertyChangeChannel(golem, nameof(IGolem.CurrentJob), (Job? currentJob) =>
+            {
+                jobStatusChannel = PropertyChangeChannel(currentJob, nameof(currentJob.Status),
+                    (JobStatus v) => _logger.LogInformation($"Current job Status update: {v}"));
+
+            });
+
+            // Then
+            var startTask = golem.Start();
+            Assert.Equal(GolemStatus.Starting, await TestUtils.ReadChannel<GolemStatus?>(statusChannel));
+            await startTask;
+            Assert.Equal(GolemStatus.Ready, await TestUtils.ReadChannel<GolemStatus?>(statusChannel));
+
+            _logger.LogInformation("Starting Sample App");
+            var app = _requestor?.CreateSampleApp() ?? throw new Exception("Requestor not started yet");
+            Assert.True(app.Start());
+
+            // Await for current job to be in Computing state
+            Job? currentJob = await ReadChannel<Job?>(jobChannel);
+            var currentState = await ReadChannel(jobStatusChannel, (JobStatus s) => s != JobStatus.Computing, 30_000);
+
+            // Access Provider process
+            var providerPidFile = Path.Combine(golemPath, "modules", "golem-data", "provider", "ya-provider.pid");
+            var providerPidTxt = await TestUtils.WaitForFileAndRead(providerPidFile);
+            var providerPid = Int32.Parse(providerPidTxt);
+            var providerProcess = Process.GetProcessById(providerPid);
+
+            var subprocesses = FindSubprocesses(providerPid);
+            Assert.Contains("ya-runtime-ai.exe", RunningExecutablesNames(subprocesses));
```

Comment:
On Windows this list does not include `dummy.exe` and instead there are only `ya-runtime-ai.exe`  and `conhost.exe` (because `ya-runtime-ai.exe` is spawned without reusing parent console)